src/Artsy/Relay/createRelaySSREnvironment.ts

```diff
@@ -76,63 +76,69 @@ export function createRelaySSREnvironment(config: Config = {}) {
 
   const network =
     relayNetwork ||
-    new RelayNetworkLayer([
-      // TODO: Better introspection around if this is a SearchBar query,
-      // or further refactoring to extract `addMiddlewareToEnvironment(environment)`,
-      // to be used in the SearchBar QueryRenderer (for example).
-      next => req => {
-        if (req.id === "SearchBarSuggestQuery" && req.variables.term === "")
-          return Promise.resolve({ data: { viewer: {} } })
-        return next(req)
-      },
-      urlMiddleware({
-        url: METAPHYSICS_ENDPOINT,
-        headers: !!user
-          ? {
-              ...headers,
-              "X-USER-ID": user && user.id,
-              "X-ACCESS-TOKEN": user && user.accessToken,
+    new RelayNetworkLayer(
+      [
+        // TODO: Better introspection around if this is a SearchBar query,
+        // or further refactoring to extract `addMiddlewareToEnvironment(environment)`,
+        // to be used in the SearchBar QueryRenderer (for example).
+        next => req => {
+          if (req.id === "SearchBarSuggestQuery" && req.variables.term === "")
+            return Promise.resolve({ data: { viewer: {} } })
+          return next(req)
+        },
+        urlMiddleware({
+          url: METAPHYSICS_ENDPOINT,
+          headers: !!user
+            ? {
+                ...headers,
+                "X-USER-ID": user && user.id,
+                "X-ACCESS-TOKEN": user && user.accessToken,
+              }
+            : headers,
+        }),
+        relaySSRMiddleware.getMiddleware(),
+        cacheMiddleware({
+          size: 100, // max 100 requests
+          ttl: 900000, // 15 minutes
+          onInit: queryResponseCache => {
+            if (!isServer) {
+              hydrateCacheFromSSR(queryResponseCache)
             }
-          : headers,
-      }),
-      relaySSRMiddleware.getMiddleware(),
-      cacheMiddleware({
-        size: 100, // max 100 requests
-        ttl: 900000, // 15 minutes
-        onInit: queryResponseCache => {
-          if (!isServer) {
-            hydrateCacheFromSSR(queryResponseCache)
+          },
+        }),
+        next => async req => {
+          const res = await next(req)
+          const httpError = get(
+            res,
+            r => r.json.extensions.principalField.httpStatusCode
+          )
+          if (httpError) {
+            throw new HttpError(httpError)
+          } else {
+            return res
+          }
+        },
```

Comment:
Can this be moved into a function in the [middleware](https://github.com/artsy/reaction/tree/master/src/Artsy/Relay/middleware) folder? 

(And if there's any other custom stuff in here that can be moved, that would be good too.)